app/src/main/java/com/github/se/project/ui/lesson/AddLesson.kt

```diff
@@ -0,0 +1,239 @@
+package com.github.se.project.ui.lesson
+
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.widget.Toast
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.RangeSlider
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.sample.model.lesson.Lesson
+import com.android.sample.model.lesson.LessonStatus
+import com.android.sample.model.lesson.LessonsViewModel
+import com.github.se.project.model.profile.Language
+import com.github.se.project.model.profile.ListProfilesViewModel
+import com.github.se.project.model.profile.TutoringSubject
+import com.github.se.project.ui.components.WritableDropdown
+import com.github.se.project.ui.navigation.NavigationActions
+import com.github.se.project.ui.navigation.Screen
+import java.util.Calendar
+
+@Composable
+fun AddLessonScreen(
+    navigationActions: NavigationActions,
+    listProfilesViewModel: ListProfilesViewModel =
+        viewModel(factory = ListProfilesViewModel.Factory),
+    lessonViewModel: LessonsViewModel = viewModel(factory = LessonsViewModel.Factory)
+) {
+  var title by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+  var subject by remember { mutableStateOf("") }
+  var language by remember { mutableStateOf("") }
+  var minPrice by remember { mutableStateOf(0.0) }
+  var maxPrice by remember { mutableStateOf(0.0) }
+  var timeSlot by remember { mutableStateOf("") }
+  val calendar = Calendar.getInstance()
+
+  val profile = listProfilesViewModel.currentProfile.collectAsState()
+  val lessonState = lessonViewModel.lessons
```

Comment:
same comment as above for the lessonState variable!

Reply:
Done